src/environment.js

```diff
@@ -120,7 +119,11 @@ var Environment = Obj.extend({
             throw new Error('template names must be a string: ' + name);
         }
 
-        var tmpl = this.cache[name];
+        for (var i = 0; i < this.loaders.length; i++) {
+            var _name = parentName? this.loaders[i].resolve(parentName, name) : name;
+            tmpl = this.loaders[i].cache[_name];
```

Comment:
What does moving the cache into the loaders fix again? Since you are running the resolution here as well, doesn't that fix the caching problem?